src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -140,48 +172,32 @@ def make_cluster_bootstrap_script_file(region, entry_point, code, suffix):
     # to distinguish it from non assets. It will be needed to annotate the bundleDepends,
     # when the wrapper record object is no more accessible.
 
-    for dep in reversed(info["runSpec"]["bundledDepends"]):
-        file_handle = get_handler(dep["id"])
-        if isinstance(file_handle, dxpy.DXFile):
-            asset_record_id = file_handle.get_properties().get("AssetBundle")
-            asset_record = None
-            if asset_record_id:
-                asset_record = dxpy.DXRecord(asset_record_id)
-                if asset_record:
-                    try:
-                        asset_json = {"name": asset_record.describe().get("name"),
-                                              "project": asset_record.get_proj_id(),
-                                              "folder": asset_record.describe().get("folder"),
-                                              "version": asset_record.describe(fields={"properties": True}
-                                                                               )["properties"]["version"]
-                                              }
-                        if dep.get("stages"):
-                            asset_json["stages"] = dep["stages"]
-                        asset_depends.append(asset_json)
-                        deps_to_remove.append(dep)
-                    except DXError:
-                        print("Describe failed on the assetDepends record object with ID - " +
-                              asset_record_id + "\n", file=sys.stderr)
-                        pass
-            else:
-                break
-    # Reversing the order of the asset_depends[] so that original order is maintained
-    asset_depends.reverse()
-    # resources/ directory
-    created_resources_directory = False
-    if not omit_resources:
-        for dep in info["runSpec"]["bundledDepends"]:
-            if dep in deps_to_remove:
-                continue
-            handler = get_handler(dep["id"])
-            if isinstance(handler, dxpy.DXFile):
+        download_completed = omit_resources
+        deps_downloaded = set()
+        created_resources_directory = False
+
+        # Download resources from the source region      
+        for dep in info["runSpec"]["bundledDependsByRegion"][source_region]:
```

Comment:
I changed it to hit resources first, so if downloading fails we can quit it earlier. But if there is no error, it would check all deps regardless of the order we use, so the total number of calls should be the same.